Scripts/Python/plasma/Plasma.py

```diff
@@ -5120,4806 +5120,4811 @@ class ptGUIPopUpMenu:
 
     def setReflOffset(self,s, secs = 0):
         """Sets the attribute to s over secs time"""
-        pass
+        raise NotImplementedError()
 
     def setRippleScale(self,s, secs = 0):
         """Sets the attribute to s over secs time"""
-        pass
+        raise NotImplementedError()
 
     def setSpecularEnd(self,s, secs = 0):
         """Sets the attribute to s over secs time"""
-        pass
+        raise NotImplementedError()
 
     def setSpecularMute(self,s, secs = 0):
         """Sets the attribute to s over secs time"""
-        pass
+        raise NotImplementedError()
 
     def setSpecularNoise(self,s, secs = 0):
         """Sets the attribute to s over secs time"""
-        pass
+        raise NotImplementedError()
 
     def setSpecularStart(self,s, secs = 0):
         """Sets the attribute to s over secs time"""
-        pass
+        raise NotImplementedError()
 
     def setSpecularTint(self,s, secs = 0):
         """Sets the attribute to s over secs time"""
-        pass
+        raise NotImplementedError()
 
     def setTexAmpOverLen(self,s, secs = 0):
         """Sets the attribute to s over secs time"""
-        pass
+        raise NotImplementedError()
 
     def setTexAngleDev(self,s, secs = 0):
         """Sets the attribute to s over secs time"""
-        pass
+        raise NotImplementedError()
 
     def setTexChop(self,s, secs = 0):
         """Sets the attribute to s over secs time"""
-        pass
+        raise NotImplementedError()
 
     def setTexMaxLength(self,s, secs = 0):
         """Sets the attribute to s over secs time"""
-        pass
+        raise NotImplementedError()
 
     def setTexMinLength(self,s, secs = 0):
         """Sets the attribute to s over secs time"""
-        pass
+        raise NotImplementedError()
 
     def setWaterHeight(self,s, secs = 0):
         """Sets the attribute to s over secs time"""
-        pass
+        raise NotImplementedError()
 
     def setWaterOffset(self,s, secs = 0):
         """Sets the attribute to s over secs time"""
-        pass
+        raise NotImplementedError()
 
     def setWaterOpacity(self,s, secs = 0):
         """Sets the attribute to s over secs time"""
-        pass
+        raise NotImplementedError()
 
     def setWaterTint(self,s, secs = 0):
         """Sets the attribute to s over secs time"""
-        pass
+        raise NotImplementedError()
 
     def setWaveFalloff(self,s, secs = 0):
         """Sets the attribute to s over secs time"""
-        pass
+        raise NotImplementedError()
 
     def setWaveOffset(self,s, secs = 0):
         """Sets the attribute to s over secs time"""
-        pass
+        raise NotImplementedError()
 
     def setWindDir(self,s, secs = 0):
         """Sets the attribute to s over secs time"""
-        pass
+        raise NotImplementedError()
 
+
+try:
+    from Plasma import *  # builtin exists if if we are in the engine
+except ImportError:
+    pass  # fallback to stubs for code analysis
```

Comment:
I think I have a slight preference for this handling to be done in __init__.py by:
```python
try:
    from Plasma import *
except ImportError:
    from .Plasma import *
```

I am assuming that Python can correctly tell the difference between Plasma and .Plasma however :/

Reply:
I made a new branch using this idea, see [here](https://github.com/aqua-uru/Plasma/blob/plasma-module-single/Scripts/Python/plasma/__init__.py). It seems to work fine in-game and the code analysis works too (at least as well as the first branch does on my machine), except for the file `PlasmaTypes.py`. The problem there is that it imports the `plasma` package because it uses classes from `Plasma.py` while it is also itself part of the `plasma` package now. This causes pylint to show a 'function already defined' error (though it seems no problem in-game).

Any ideas on how to deal with that? I thought about having it import `Plasma.py` specifically to avoid the recurrence, but that would require a try-catch to make sure it uses the right version (in-engine or stubs). Maybe have a `PlasmaStubsLoader.py` or something that does the try-catch for all maybe-stubs files, and have `from .PlasmaStubsLoader import *` in both `PlasmaTypes.py` and `__init__.py`?

There's also this line in `pch.py`:

```python
if not hasattr(Plasma,name) and not hasattr(PlasmaTypes,name):
```

which should probably be changed to `if not hasattr(plasma, name)`.

Reply:
I added the `PlasmaLoader.py` to the `plasma-module-single` branch, and made the change in `pch.py`. All seems to be working well! Do you prefer the `single` branch to this one (using a single `plasma` import rather than the original separate ones? If so, I will use it to replace this one.

Reply:
I can't find the "go to definition" button. But, when I hover over `ptAgeVault()` in Ahnonay.py, it helpfully displays the docstring. I'm liking the `plasma-module-single` branch--of course, other opinions are always welcome.

Before we merge any of this, we need to be sure to test that running this PR through plPythonPack works as expected.